### PR TITLE
Use `User#permissions` to check permissions

### PR DESCRIPTION
**Pull request by mansakondo**
This PR refactors all the occurences of `Business::Permission.new(user.subscriptions)` with `user.permissions`.

### CODE REVIEW

File: app/components/businesses/hiring_fee_component.rb [hunk 1]
```diff
@@ -8,7 +8,7 @@ def initialize(user, conversation)
     end
 
     def render?
-      Businesses::Permission.new(user.subscriptions).pays_hiring_fee? &&
+      user.permissions.pays_hiring_fee? &&
         conversation.hiring_fee_eligible?
```
Comment (joemasilotti) [inline app/components/businesses/hiring_fee_component.rb:12]:
```suggestion
      user.permissions.pays_hiring_fee? && conversation.hiring_fee_eligible?
```

File: app/components/businesses/upgrade_required_component.rb [hunk 2]
```diff
@@ -11,7 +11,7 @@ def render?
     end
 
     def expired?
-      !permission.active_subscription?
+      !permissions.active_subscription?
```
Comment (joemasilotti) [inline app/components/businesses/upgrade_required_component.rb:14]:
```suggestion
      !user.permissions.active_subscription?
```

Then we can remove the private helper method, too.

File: app/policies/message_policy.rb [hunk 3]
```diff
@@ -5,7 +5,7 @@ def create?
     elsif developer_sender?
       true
     elsif business_sender?
-      permission.can_message_developer?(role_type: developer.role_type)
+      permissions.can_message_developer?(role_type: developer.role_type)
```
Comment (joemasilotti) [inline app/policies/message_policy.rb:8]:
```suggestion
      user.permissions.can_message_developer?(role_type: developer.role_type)
```

And then we can remove the private helper method, too.

**Review by joemasilotti — CHANGES_REQUESTED** [review]
Thanks for tackling this, too! I made a few code suggestions which remove some private helper methods we no longer need. Once those are updated this is good to merge.

**Comment by mansakondo** [comment]
> Thanks for tackling this, too! I made a few code suggestions which remove some private helper methods we no longer need. Once those are updated this is good to merge.

No problem. I committed the changes you requested.

**Comment by joemasilotti** [comment]
Looks great - thanks for the changes! I need to do a bit more manual testing. But outside of that, can you replace the tests checking for false to `refute`? That’s a bit more concise. Thanks again!

**Comment by mansakondo** [comment]
> Looks great - thanks for the changes! I need to do a bit more manual testing. But outside of that, can you replace the tests checking for false to `refute`? That’s a bit more concise. Thanks again!

Happy to help! Sorry, which tests are you referring to?

**Comment by joemasilotti** [comment]
Sorry, disregard that comment. It was actually for another PR! So much work is going on right now I'm getting some wires crossed.

**Comment by mansakondo** [comment]
>



> Sorry, disregard that comment. It was actually for another PR! So much work is going on right now I'm getting some wires crossed.

No worries :)

**Pull request merged**